### PR TITLE
Users can now search for products

### PR DESCRIPTION
**Pull request by samakshjain**
[#126384775]

### CODE REVIEW

File: shop/templates/utils.html [hunk 1]
```diff
@@ -20,7 +20,7 @@
       {% set name = listing.product.name %}
     {% endif %}
   {% endif %}
-  <a href="{{ url_for('products.product', handle=listing.product_identifier) }}"
+  <a href="{{ url_for('products.product', handle=listing.product_identifier or '404') }}"
```
Comment (prakashpp) [inline shop/templates/utils.html]: remove

File: shop/public/views.py [hunk 2]
```diff
@@ -81,6 +83,25 @@ def register():
...
+    q = request.args.get('q')
+    page = request.args.get('page', type=int) or None
+    per_page = request.args.get('per_page', type=int) or None
+    shop_query = ChannelListing.get_shop_query().filter_by_domain(
+        [
+            ('product.rec_name', 'ilike', '%{}%'.format(q)),
+            ('channel', '=', current_channel.id)
+        ],
+    )
+    paginate = shop_query.paginate(page=page, per_page=per_page)
```
Comment (sharoonthomas) [inline shop/public/views.py]: results = ..

File: shop/public/views.py [hunk 3]
```diff
@@ -81,6 +83,25 @@ def register():
...
+    return render_template('public/search.html', paginate=paginate)
```
Comment (sharoonthomas) [inline shop/public/views.py]: results=results

File: shop/templates/public/search.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        </li>
+        {% else %}
+        <li>
+          <p class="text-center"><strong>We found nothing!</strong></p>
+        </li>
+      {% endfor %}
+    </ul>
+  </section>
+  {% if results.pages > 1 %}
+    {{ render_pagination(paginate, endpoint='public.search', q=query) }}
```
Comment (prakashpp) [inline shop/templates/public/search.html]: how are you getting paginate in the template?

Comment (samakshjain) [inline shop/templates/public/search.html]: Oops. Missed it.

**Pull request merged**